src/ai/defaultai.cc

```diff
@@ -269,13 +267,34 @@ void DefaultAI::think() {
 
 	SchedulerTaskId due_task = SchedulerTaskId::kUnset;
 
-	sort_task_pool();
+	// we need to get more one most urgent job
+	sort_task_pool(1);
 
 	const int32_t delay_time = gametime.get() - taskPool.front().due_time.get();
 
-	// Here we decide how many jobs will be run now (none - 5)
-	// in case no job is due now, it can be zero
-	uint32_t jobs_to_run_count = (delay_time < 0) ? 0 : 1;
+	// This portion of code tries to keep the speed of game high, but so that AI's jobs
+	// have reasonably enough time for execution and are not too much delayed
+	if (game().is_auto_speed()) {
+		int32_t speed_diff = 0;
+		if (delay_time > 4000) {
+			speed_diff = -100;
+		} else if (delay_time < 1000) {
+			speed_diff = +100;
+		}
+		if (speed_diff != 0) {
+			if (GameController* const ctrl = game().game_controller()) {
+				if ((ctrl->desired_speed() > 950 && ctrl->desired_speed() < 30000) ||
+				    (ctrl->desired_speed() < 1000 && speed_diff > 0) ||
+				    (ctrl->desired_speed() > 29999 && speed_diff < 0)) {
+					ctrl->set_desired_speed(ctrl->desired_speed() + speed_diff);
+				}
+			}
+		}
+	}
+
+	// Here we decide limit for jobs will be run now
+	// generelly 3, if situation is wrong: 4
+	uint32_t jobs_to_run_count = std::min<uint32_t>((delay_time > 3000) ? 5 : 3, taskPool.size());
```

Comment:
```suggestion
	uint32_t jobs_to_run_count = std::min<uint32_t>((delay_time > 3000) ? 5 : 3, taskPool.size());
```
        uint32_t jobs_to_run_count = std::min<uint32_t>(delay_time / 2000 + 1, taskPool.size() / 2);